sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/AggregationTemporalitySelector.java

```diff
@@ -49,6 +49,29 @@ static AggregationTemporalitySelector deltaPreferred() {
     };
   }
 
+  /**
+   * A common implementation of {@link AggregationTemporalitySelector} which reduces memory.
+   *
+   * <p>{@link AggregationTemporality#DELTA} is returned for {@link InstrumentType#COUNTER} and
+   * {@link InstrumentType#HISTOGRAM}. {@link AggregationTemporality#CUMULATIVE} is returned for
+   * {@link InstrumentType#UP_DOWN_COUNTER}, {@link InstrumentType#OBSERVABLE_UP_DOWN_COUNTER}, and
+   * {@link InstrumentType#OBSERVABLE_COUNTER}.
+   */
+  static AggregationTemporalitySelector lowMemory() {
+    return instrumentType -> {
+      switch (instrumentType) {
+        case UP_DOWN_COUNTER:
+        case OBSERVABLE_UP_DOWN_COUNTER:
+        case OBSERVABLE_COUNTER:
+          return AggregationTemporality.CUMULATIVE;
+        case COUNTER:
+        case HISTOGRAM:
+        default:
+          return AggregationTemporality.DELTA;
+      }
```

Comment:
Can you add test coverage for this in `AggregationTemporalitySelectorTest`? Seems silly to write a test that essentially just restates the code, but makes it a bit harder for a change to this definition to accidentally slip by.